extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelContextRecorder.java

```diff
@@ -107,4 +108,34 @@ public void registerManagementEventBridge(RuntimeValue<CamelContext> camelContex
                 .getManagementStrategy()
                 .addEventNotifier(new CamelManagementEventBridge(observedManagementEvents));
     }
+
+    public RuntimeValue<CamelContextCustomizer> createBacklogTracerCustomizer(CamelConfig config) {
+        return new RuntimeValue<>(context -> {
+            // must enable source location so tracer tooling knows to map breakpoints to source code
+            context.setSourceLocationEnabled(true);
+
+            // enable tracer on camel
+            context.setBacklogTracing(config.trace.enabled);
+            context.setBacklogTracingStandby(config.trace.standby);
+            context.setBacklogTracingTemplates(config.trace.traceTemplates);
+
+            BacklogTracer tracer = BacklogTracer.createTracer(context);
+            tracer.setEnabled(config.trace.enabled);
+            tracer.setStandby(config.trace.standby);
+            tracer.setBacklogSize(config.trace.backlogSize);
+            tracer.setRemoveOnDump(config.trace.removeOnDump);
+            tracer.setBodyMaxChars(config.trace.bodyMaxChars);
+            tracer.setBodyIncludeStreams(config.trace.bodyIncludeStreams);
+            tracer.setBodyIncludeFiles(config.trace.bodyIncludeFiles);
+            tracer.setIncludeExchangeProperties(config.trace.includeExchangeProperties);
+            tracer.setIncludeExchangeVariables(config.trace.includeExchangeVariables);
+            tracer.setIncludeException(config.trace.includeException);
+            tracer.setTraceRests(config.trace.traceRests);
+            tracer.setTraceTemplates(config.trace.traceTemplates);
+            tracer.setTracePattern(config.trace.tracePattern.orElse(""));
```

Comment:
these should be null instead of empty string

Reply:
The same with `setTraceFilter`. it should be `null` as well?
```java
tracer.setTraceFilter(config.trace.traceFilter.orElse(""));
```

Reply:
@davsclaus set with `null` on both of them.

Reply:
yes both